diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/component/scroll.ets b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/component/scroll.ets
index b223717501..a6762a81f4 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/component/scroll.ets
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/component/scroll.ets
@@ -86,6 +86,9 @@ export class Scroller implements MaterializedBase {
     public currentOffset(): OffsetResult | undefined {
         return this.currentOffset_serialize()
     }
+    public offset(): OffsetResult | undefined {
+        return this.offset_serialize()
+    }
     public scrollToIndex(value: int32, smooth?: boolean, align?: ScrollAlign, options?: ScrollToIndexOptions): void {
         const value_casted = value as (int32)
         const smooth_casted = smooth as (boolean | undefined)
@@ -150,6 +153,17 @@ export class Scroller implements MaterializedBase {
         const returnResult : OffsetResult | undefined = buffer
         return returnResult
     }
+    offset_serialize(): OffsetResult | undefined {
+        const retval  = ArkUIGeneratedNativeModule._Scroller_offset(this.peer!.ptr)
+        let retvalDeserializer : DeserializerBase = new DeserializerBase(retval, retval.length)
+        const buffer_runtimeType = retvalDeserializer.readInt8().toInt()
+        let buffer : OffsetResult | undefined = undefined
+        if ((buffer_runtimeType) != (RuntimeType.UNDEFINED)) {
+            buffer = OffsetResult_serializer.read(retvalDeserializer)
+        }
+        const returnResult : OffsetResult | undefined = buffer
+        return returnResult
+    }
     scrollToIndex_serialize(value: int32, smooth?: boolean, align?: ScrollAlign, options?: ScrollToIndexOptions): void {
         const thisSerializer : SerializerBase = SerializerBase.hold()
         if (smooth !== undefined) {
diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/framework/arkts/ArkUIGeneratedNativeModule.ets b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/framework/arkts/ArkUIGeneratedNativeModule.ets
index b0fb6680ae..903a800f95 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/framework/arkts/ArkUIGeneratedNativeModule.ets
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/arkui-ohos/generated/framework/arkts/ArkUIGeneratedNativeModule.ets
@@ -5425,6 +5425,7 @@ export class ArkUIGeneratedNativeModule {
     native static _Scroller_fling(ptr: KPointer, velocity: KDouble): void
     native static _Scroller_scrollPage(ptr: KPointer, thisArray: KSerializerBuffer, thisLength: int32): void
     native static _Scroller_currentOffset(ptr: KPointer): KInteropReturnBuffer
+    native static _Scroller_offset(ptr: KPointer): KInteropReturnBuffer
     native static _Scroller_scrollToIndex(ptr: KPointer, value: KInt, thisArray: KSerializerBuffer, thisLength: int32): void
     native static _Scroller_scrollBy(ptr: KPointer, thisArray: KSerializerBuffer, thisLength: int32): void
     native static _Scroller_isAtEnd(ptr: KPointer): boolean
diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/arkoala_api_generated.h b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/arkoala_api_generated.h
index ca1e5cf81f..3440b010d0 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/arkoala_api_generated.h
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/arkoala_api_generated.h
@@ -27174,6 +27174,7 @@ typedef struct GENERATED_ArkUIScrollerAccessor {
                        const Ark_ScrollPageOptions* value);
     Opt_OffsetResult (*currentOffset)(Ark_VMContext vmContext,
                                       Ark_Scroller peer);
+    Opt_OffsetResult (*offset)(Ark_Scroller peer);
     void (*scrollToIndex)(Ark_VMContext vmContext,
                           Ark_Scroller peer,
                           Ark_Int32 value,
diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/bridge_generated.cc b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/bridge_generated.cc
index de520deca0..b2edb48cf1 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/bridge_generated.cc
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/bridge_generated.cc
@@ -35359,6 +35359,20 @@ KInteropReturnBuffer impl_Scroller_currentOffset(KVMContext vmContext, Ark_Nativ
         return _retSerializer.toReturnBuffer();
 }
 KOALA_INTEROP_CTX_1(Scroller_currentOffset, KInteropReturnBuffer, Ark_NativePointer)
+KInteropReturnBuffer impl_Scroller_offset(Ark_NativePointer thisPtr) {
+        Ark_Scroller self = reinterpret_cast<Ark_Scroller>(thisPtr);
+        const auto &retValue = GetAccessors()->getScrollerAccessor()->offset(self);
+        SerializerBase _retSerializer {};
+        if (retValue.tag != INTEROP_TAG_UNDEFINED) {
+            _retSerializer.writeInt8(INTEROP_RUNTIME_OBJECT);
+            const auto retValueTmpValue = retValue.value;
+            OffsetResult_serializer::write(_retSerializer, retValueTmpValue);
+        } else {
+            _retSerializer.writeInt8(INTEROP_RUNTIME_UNDEFINED);
+        }
+        return _retSerializer.toReturnBuffer();
+}
+KOALA_INTEROP_1(Scroller_offset, KInteropReturnBuffer, Ark_NativePointer)
 void impl_Scroller_scrollToIndex(KVMContext vmContext, Ark_NativePointer thisPtr, Ark_Int32 value, KSerializerBuffer thisArray, int32_t thisLength) {
         Ark_Scroller self = reinterpret_cast<Ark_Scroller>(thisPtr);
         DeserializerBase thisDeserializer(thisArray, thisLength);
diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/dummy_impl.cc b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/dummy_impl.cc
index 4f6f1ce296..ff13ec8428 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/dummy_impl.cc
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/dummy_impl.cc
@@ -39771,6 +39771,18 @@ namespace OHOS::Ace::NG::GeneratedModifier {
         appendGroupedLog(1, out);
         return {};
     }
+    Opt_OffsetResult OffsetImpl(Ark_Scroller peer)
+    {
+        if (!needGroupedLog(1))
+        {
+            return {};
+        }
+        string out("offset(");
+        out.append(") \n");
+        out.append("[return {}] \n");
+        appendGroupedLog(1, out);
+        return {};
+    }
     void ScrollToIndexImpl(Ark_VMContext vmContext,
                            Ark_Scroller peer,
                            Ark_Int32 value,
@@ -47670,6 +47682,7 @@ namespace OHOS::Ace::NG::GeneratedModifier {
             ScrollerAccessor::FlingImpl,
             ScrollerAccessor::ScrollPageImpl,
             ScrollerAccessor::CurrentOffsetImpl,
+            ScrollerAccessor::OffsetImpl,
             ScrollerAccessor::ScrollToIndexImpl,
             ScrollerAccessor::ScrollByImpl,
             ScrollerAccessor::IsAtEndImpl,
diff --git a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/real_impl.cc b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/real_impl.cc
index 1d762e5cb5..f406abe240 100644
--- a/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/real_impl.cc
+++ b/frameworks/bridge/arkts_frontend/koala_projects/arkoala-arkts/framework/native/src/generated/real_impl.cc
@@ -16603,7 +16603,11 @@ namespace OHOS::Ace::NG::GeneratedModifier {
     {
     }
     Opt_OffsetResult CurrentOffsetImpl(Ark_VMContext vmContext,
-                                       Ark_Scroller peer)
+                                    Ark_Scroller peer)
+    {
+        return {};
+    }
+    Opt_OffsetResult OffsetImpl(Ark_Scroller peer)
     {
         return {};
     }
@@ -21160,6 +21164,7 @@ namespace OHOS::Ace::NG::GeneratedModifier {
             ScrollerAccessor::FlingImpl,
             ScrollerAccessor::ScrollPageImpl,
             ScrollerAccessor::CurrentOffsetImpl,
+            ScrollerAccessor::OffsetImpl,
             ScrollerAccessor::ScrollToIndexImpl,
             ScrollerAccessor::ScrollByImpl,
             ScrollerAccessor::IsAtEndImpl,
diff --git a/frameworks/bridge/declarative_frontend/jsview/js_scroller.cpp b/frameworks/bridge/declarative_frontend/jsview/js_scroller.cpp
index f89e492dbb..0a602baa4c 100644
--- a/frameworks/bridge/declarative_frontend/jsview/js_scroller.cpp
+++ b/frameworks/bridge/declarative_frontend/jsview/js_scroller.cpp
@@ -72,6 +72,7 @@ void JSScroller::JSBind(BindingTarget globalObj)
     JSClass<JSScroller>::CustomMethod("fling", &JSScroller::Fling);
     JSClass<JSScroller>::CustomMethod("scrollPage", &JSScroller::ScrollPage);
     JSClass<JSScroller>::CustomMethod("currentOffset", &JSScroller::CurrentOffset);
+    JSClass<JSScroller>::CustomMethod("offset", &JSScroller::Offset);
     JSClass<JSScroller>::CustomMethod("scrollToIndex", &JSScroller::ScrollToIndex);
     JSClass<JSScroller>::CustomMethod("scrollBy", &JSScroller::ScrollBy);
     JSClass<JSScroller>::CustomMethod("isAtEnd", &JSScroller::IsAtEnd);
@@ -308,6 +309,22 @@ void JSScroller::CurrentOffset(const JSCallbackInfo& args)
     args.SetReturnValue(retObj);
 }
 
+void JSScroller::Offset(const JSCallbackInfo& args)
+{
+    auto scrollController = controllerWeak_.Upgrade();
+    if (!scrollController) {
+        EventReport::ReportScrollableErrorEvent("Scroller", ScrollableErrorType::CONTROLLER_NOT_BIND,
+            "The controller does not bind a component when calling Offset function");
+        return;
+    }
+    auto retObj = JSRef<JSObject>::New();
+    ContainerScope scope(instanceId_);
+    auto offset = scrollController->GetCurrentOffset();
+    retObj->SetProperty("xOffset", offset.GetX());
+    retObj->SetProperty("yOffset", offset.GetY());
+    args.SetReturnValue(retObj);
+}
+
 void JSScroller::ScrollBy(const JSCallbackInfo& args)
 {
     if (args.Length() < 2) {
diff --git a/frameworks/bridge/declarative_frontend/jsview/js_scroller.h b/frameworks/bridge/declarative_frontend/jsview/js_scroller.h
index 9511ba97ff..6228512620 100644
--- a/frameworks/bridge/declarative_frontend/jsview/js_scroller.h
+++ b/frameworks/bridge/declarative_frontend/jsview/js_scroller.h
@@ -38,6 +38,7 @@ public:
     void Fling(const JSCallbackInfo& args);
     void ScrollPage(const JSCallbackInfo& args);
     void CurrentOffset(const JSCallbackInfo& args);
+    void Offset(const JSCallbackInfo& args);
     void ScrollToIndex(const JSCallbackInfo& args);
     void ScrollBy(const JSCallbackInfo& args);
     void IsAtEnd(const JSCallbackInfo& args);
diff --git a/frameworks/core/interfaces/native/generated/interface/arkoala_api_generated.h b/frameworks/core/interfaces/native/generated/interface/arkoala_api_generated.h
index cd1f71ae8d..4dc3dc6f9c 100644
--- a/frameworks/core/interfaces/native/generated/interface/arkoala_api_generated.h
+++ b/frameworks/core/interfaces/native/generated/interface/arkoala_api_generated.h
@@ -27188,6 +27188,7 @@ typedef struct GENERATED_ArkUIScrollerAccessor {
                        const Ark_ScrollPageOptions* value);
     Opt_OffsetResult (*currentOffset)(Ark_VMContext vmContext,
                                       Ark_Scroller peer);
+    Opt_OffsetResult (*offset)(Ark_Scroller peer);
     void (*scrollToIndex)(Ark_VMContext vmContext,
                           Ark_Scroller peer,
                           Ark_Int32 value,
diff --git a/frameworks/core/interfaces/native/implementation/scroller_accessor.cpp b/frameworks/core/interfaces/native/implementation/scroller_accessor.cpp
index a40fc231bd..dd310b3755 100644
--- a/frameworks/core/interfaces/native/implementation/scroller_accessor.cpp
+++ b/frameworks/core/interfaces/native/implementation/scroller_accessor.cpp
@@ -70,6 +70,13 @@ Opt_OffsetResult CurrentOffsetImpl(Ark_VMContext vmContext,
     CHECK_NULL_RETURN(peer, invalid);
     return peer->TriggerCurrentOffset(reinterpret_cast<ani_env*>(vmContext));
 }
+Opt_OffsetResult OffsetImpl(Ark_Scroller peer)
+{
+    auto invalid = Converter::ArkValue<Opt_OffsetResult>();
+    CHECK_NULL_RETURN(peer, invalid);
+    return peer->TriggerOffset();
+}
+
 void ScrollToIndexImpl(Ark_VMContext vmContext,
                        Ark_Scroller peer,
                        Ark_Int32 value,
@@ -127,6 +134,7 @@ const GENERATED_ArkUIScrollerAccessor* GetScrollerAccessor()
         ScrollerAccessor::FlingImpl,
         ScrollerAccessor::ScrollPageImpl,
         ScrollerAccessor::CurrentOffsetImpl,
+        ScrollerAccessor::OffsetImpl,
         ScrollerAccessor::ScrollToIndexImpl,
         ScrollerAccessor::ScrollByImpl,
         ScrollerAccessor::IsAtEndImpl,
diff --git a/frameworks/core/interfaces/native/implementation/scroller_peer_impl.cpp b/frameworks/core/interfaces/native/implementation/scroller_peer_impl.cpp
index 95dee6d45b..caea048a14 100644
--- a/frameworks/core/interfaces/native/implementation/scroller_peer_impl.cpp
+++ b/frameworks/core/interfaces/native/implementation/scroller_peer_impl.cpp
@@ -235,13 +235,28 @@ Opt_OffsetResult ScrollerPeerImpl::TriggerCurrentOffset(ani_env* env)
     auto scrollController = controllerWeak_.Upgrade();
     if (!scrollController) {
         LOGE("ARKOALA ScrollerPeerImpl::TriggerCurrentOffset Controller not bound to component.");
-        ScrollerPeerImpl::ThrowControllerError(env);
+        LOGE("dlh revisedTriggerCurrentOffset");
+        // ScrollerPeerImpl::ThrowControllerError(env);
         return Converter::ArkValue<Opt_OffsetResult>(Offset());
     }
     ContainerScope scope(instanceId_);
     auto offset = scrollController->GetCurrentOffset(); // the result of GetCurrentOffset need to be returned
     return Converter::ArkValue<Opt_OffsetResult>(offset);
 }
+Opt_OffsetResult ScrollerPeerImpl::TriggerOffset()
+{
+    auto scrollController = controllerWeak_.Upgrade();
+    if (!scrollController) {
+        LOGE("ARKOALA ScrollerPeerImpl::TriggerOffset Controller not bound to component.");
+        return (Opt_OffsetResult) {
+            .tag = static_cast<InteropTag>(INTEROP_TAG_UNDEFINED),
+            .value = { 0, 0 }
+            };
+    }
+    ContainerScope scope(instanceId_);
+    auto offset = scrollController->GetCurrentOffset(); // the result of GetCurrentOffset need to be returned
+    return Converter::ArkValue<Opt_OffsetResult>(offset);
+}
 
 void ScrollerPeerImpl::TriggerScrollToIndex(ani_env* env, const Ark_Int32 value, const Opt_Boolean* smoothValue,
     const Opt_ScrollAlign* alignValue, const Opt_ScrollToIndexOptions* options)
@@ -249,7 +264,7 @@ void ScrollerPeerImpl::TriggerScrollToIndex(ani_env* env, const Ark_Int32 value,
     auto scrollController = controllerWeak_.Upgrade();
     if (!scrollController) {
         LOGE("ARKOALA ScrollerPeerImpl::TriggerScrollToIndex Controller not bound to component.");
-        ScrollerPeerImpl::ThrowControllerError(env);
+        // ScrollerPeerImpl::ThrowControllerError(env);
         return;
     }
 
diff --git a/frameworks/core/interfaces/native/implementation/scroller_peer_impl.h b/frameworks/core/interfaces/native/implementation/scroller_peer_impl.h
index 528f1df98e..0516582e52 100644
--- a/frameworks/core/interfaces/native/implementation/scroller_peer_impl.h
+++ b/frameworks/core/interfaces/native/implementation/scroller_peer_impl.h
@@ -37,6 +37,7 @@ public:
     void TriggerScrollPage0(ani_env* env, const Ark_ScrollPageOptions* value);
     void TriggerScrollPage1(bool next);
     Opt_OffsetResult TriggerCurrentOffset(ani_env* env);
+    Opt_OffsetResult TriggerOffset();
     void TriggerScrollToIndex(ani_env* env, const Ark_Int32 value, const Opt_Boolean* smooth,
         const Opt_ScrollAlign* align, const Opt_ScrollToIndexOptions* options);
     void TriggerScrollBy(ani_env* env, const Dimension& xOffset, const Dimension& yOffset);
