diff --git a/bundle.json b/bundle.json
index 9761eefbad..1e330c7099 100644
--- a/bundle.json
+++ b/bundle.json
@@ -52,6 +52,7 @@
                 "os_account",
                 "graphic_2d",
                 "graphic_3d",
+                "graphic_surface",
                 "safwk",
                 "common_event_service",
                 "data_share",
diff --git a/frameworks/core/components/camera/BUILD.gn b/frameworks/core/components/camera/BUILD.gn
index 6cd598e5e5..98ccdf03f3 100644
--- a/frameworks/core/components/camera/BUILD.gn
+++ b/frameworks/core/components/camera/BUILD.gn
@@ -48,7 +48,7 @@ if (enable_camera_framework && enable_player_framework) {
       external_deps = [
         "c_utils:utils",
         "camera_framework:camera_framework",
-        "graphic_2d:surface",
+        "graphic_surface:surface",
         "ipc:ipc_core",
         "player_framework:media_client",
       ]
diff --git a/frameworks/core/components/xcomponent/xcomponent_element.h b/frameworks/core/components/xcomponent/xcomponent_element.h
index ed98cd6116..b2eeb8f295 100644
--- a/frameworks/core/components/xcomponent/xcomponent_element.h
+++ b/frameworks/core/components/xcomponent/xcomponent_element.h
@@ -23,7 +23,7 @@
 
 #ifdef OHOS_STANDARD_SYSTEM
 #include "display_type.h"
-#include "foundation/graphic/graphic_2d/interfaces/inner_api/surface/window.h"
+#include "foundation/graphic/graphic_surface/interfaces/inner_api/surface/window.h"
 #include "foundation/window/window_manager/interfaces/innerkits/wm/window.h"
 #include "render_service_client/core/ui/rs_node.h"
 #include "render_service_client/core/ui/rs_surface_node.h"
diff --git a/test/unittest/interfaces/BUILD.gn b/test/unittest/interfaces/BUILD.gn
index 40423b00ea..53434bb248 100644
--- a/test/unittest/interfaces/BUILD.gn
+++ b/test/unittest/interfaces/BUILD.gn
@@ -124,7 +124,7 @@ ohos_unittest("form_render_test") {
     "c_utils:utils",
     "eventhandler:libeventhandler",
     "form_fwk:form_manager",
-    "graphic_2d:surface",
+    "graphic_surface:surface",
     "hilog:libhilog",
     "input:libmmi-client",
     "ipc:ipc_core",
