diff --git a/api/@internal/component/ets/column.d.ts b/api/@internal/component/ets/column.d.ts
index 96457a9ee..593121204 100644
--- a/api/@internal/component/ets/column.d.ts
+++ b/api/@internal/component/ets/column.d.ts
@@ -84,7 +84,7 @@ interface ColumnOptions {
    * @since arkts {'1.1':'11','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  space?: string | number;
+  space?: SpaceType;
 }
 
 /**
@@ -110,7 +110,7 @@ interface ColumnOptionsV2 {
    * @since arkts {'1.1':'18','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  space?: SpaceType;
+  doNotUseMe: boolean;
 }
 
 /**
@@ -207,19 +207,6 @@ interface ColumnInterface {
    * @since arkts {'1.1':'18','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  (options?: ColumnOptions): ColumnAttribute;
-  /**
-   * Set the options.
-   *
-   * @param { ColumnOptions | ColumnOptionsV2 } [options] - column options
-   * @returns { ColumnAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since arkts {'1.1':'18','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
   (options?: ColumnOptions | ColumnOptionsV2): ColumnAttribute;
 }
 
diff --git a/api/@internal/component/ets/relative_container.d.ts b/api/@internal/component/ets/relative_container.d.ts
index 1caa3043b..82659bec9 100644
--- a/api/@internal/component/ets/relative_container.d.ts
+++ b/api/@internal/component/ets/relative_container.d.ts
@@ -436,7 +436,7 @@ declare class RelativeContainerAttribute extends CommonMethod<RelativeContainerA
    * @since arkts {'1.1':'12','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  barrier(barrierStyle: Array<LocalizedBarrierStyle>): RelativeContainerAttribute;
+  // barrier(barrierStyle: Array<LocalizedBarrierStyle>): RelativeContainerAttribute;
 }
 
 /**
diff --git a/api/@internal/component/ets/row.d.ts b/api/@internal/component/ets/row.d.ts
index b3981964e..0cfac7965 100644
--- a/api/@internal/component/ets/row.d.ts
+++ b/api/@internal/component/ets/row.d.ts
@@ -109,7 +109,7 @@ interface RowOptionsV2 {
    * @since arkts {'1.1':'18','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  space?: SpaceType;
+  doNotUseMe: boolean;
 }
 
 /**
@@ -148,58 +148,6 @@ interface RowOptionsV2 {
  * @arkts 1.1&1.2
  */
 interface RowInterface {
-  /**
-   * Called when the layout is set in the horizontal direction.
-   *
-   * @param { object } value
-   * @returns { RowAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @since 7
-   */
-  /**
-   * Called when the layout is set in the horizontal direction.
-   *
-   * @param { object } value
-   * @returns { RowAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @form
-   * @since 9
-   */
-  /**
-   * Called when the layout is set in the horizontal direction.
-   *
-   * @param { object } value
-   * @returns { RowAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @since 10
-   */
-  /**
-   * Called when the layout is set in the horizontal direction.
-   *
-   * @param { object } value
-   * @returns { RowAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since 11
-   */
-  /**
-   * Called when the layout is set in the horizontal direction.
-   *
-   * Anonymous Object Rectification
-   * @param { ?RowOptions } options - row options
-   * @returns { RowAttribute }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since arkts {'1.1':'18','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  (options?: RowOptions): RowAttribute;
   /**
    * Called when the layout is set in the horizontal direction.
    *
diff --git a/api/@internal/component/ets/sidebar.d.ts b/api/@internal/component/ets/sidebar.d.ts
index d60a3cd14..17e68d541 100644
--- a/api/@internal/component/ets/sidebar.d.ts
+++ b/api/@internal/component/ets/sidebar.d.ts
@@ -21,7 +21,7 @@
 /*** if arkts 1.2 */
 import { Resource } from '../../global/resource'
 import { Length, ResourceColor, Dimension } from './units'
-import { CommonMethod, PixelMap, Bindable } from './common'
+import { CommonMethod, PixelMap, Bindable, DividerStyle } from './common'
 /*** endif */
 
 /**
@@ -554,110 +554,6 @@ interface SideBarContainerInterface {
   (type?: SideBarContainerType): SideBarContainerAttribute;
 }
 
-/**
- * Provides an interface for the style of a divider including stroke width, color, start margin
- * and end margin
- *
- * @interface DividerStyle
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @since 10
- */
-/**
- * Provides an interface for the style of a divider including stroke width, color, start margin
- * and end margin
- *
- * @interface DividerStyle
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since arkts {'1.1':'11','1.2':'20'}
- * @arkts 1.1&1.2
- */
-interface DividerStyle {
-  /**
-   * Define the stroke width of the divider
-   *
-   * @type { Length }
-   * @default 1vp
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the stroke width of the divider
-   *
-   * @type { Length }
-   * @default 1vp
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  strokeWidth: Length;
-
-  /**
-   * Define the color of the divider
-   *
-   * @type { ?ResourceColor }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the color of the divider
-   *
-   * @type { ?ResourceColor }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  color?: ResourceColor;
-
-  /**
-   * Define the start margin of the divider
-   *
-   * @type { ?Length }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the start margin of the divider
-   *
-   * @type { ?Length }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  startMargin?: Length;
-
-  /**
-   * Define the end margin of the divider
-   *
-   * @type { ?Length }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the end margin of the divider
-   *
-   * @type { ?Length }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  endMargin?: Length;
-}
-
 /**
  * The attribute function of sidebar
  *
diff --git a/api/@internal/component/ets/toggle.d.ts b/api/@internal/component/ets/toggle.d.ts
index 4f1a07f30..73b56ec38 100644
--- a/api/@internal/component/ets/toggle.d.ts
+++ b/api/@internal/component/ets/toggle.d.ts
@@ -255,7 +255,7 @@ declare interface ToggleConfiguration extends CommonConfiguration<ToggleConfigur
    * @since arkts {'1.1':'12','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  enabled: boolean;
+  toggleEnabled: boolean;
 
   /**
    * Trigger toggle select change
diff --git a/api/arkui/component/common.static.d.ets b/api/arkui/component/common.static.d.ets
index 31b466c06..e0668cde1 100644
--- a/api/arkui/component/common.static.d.ets
+++ b/api/arkui/component/common.static.d.ets
@@ -7317,13 +7317,13 @@ export declare interface SheetOptions extends BindOptions {
     /**
      * Defines sheet detents
      *
-     * @type { ?(SingleLengthDetent | DoubleLengthDetents | TripleLengthDetents) }
+     * @type { ?(TripleLengthDetents) }
      * @syscap SystemCapability.ArkUI.ArkUI.Full
      * @crossplatform
      * @atomicservice
      * @since 20
      */
-    detents?: SingleLengthDetent | DoubleLengthDetents | TripleLengthDetents;
+    detents?: TripleLengthDetents;
     /**
      * Background blur of the sheet. By default, there is no background blur.
      *
@@ -7709,7 +7709,7 @@ export declare interface SheetOptions extends BindOptions {
  * @atomicservice
  * @since 20
  */
-export type CustomStyles = (instance: CommonMethod) => void;
+export type CustomStyles = (instance: string) => void;
 /**
  * Component State Styles.
  *
@@ -10331,13 +10331,13 @@ export declare interface BackgroundOptions {
 /**
  * Defines the value of the custom property..
  *
- * @typedef { Object | undefined | Record<string, CustomProperty> } CustomProperty
+ * @typedef { Object | undefined } CustomProperty
  * @syscap SystemCapability.ArkUI.ArkUI.Full
  * @crossplatform
  * @atomicservice
  * @since 20
  */
-export declare type CustomProperty = Object | undefined | Record<string, CustomProperty>;
+export declare type CustomProperty = Object | undefined;
 export declare interface CommonMethod {
     /**
      * Sets the width of the component or its horizontal layout policy. By default, the
@@ -11076,7 +11076,7 @@ export declare interface CommonMethod {
      * @atomicservice
      * @since 20
      */
-    default onKeyEvent(event: Callback<KeyEvent, boolean> | undefined): this;
+    // default onKeyEvent(event: Callback<KeyEvent, boolean> | undefined): this;
     /**
      * Digital crown input.
      *
@@ -15346,4 +15346,4 @@ export declare function $$<T>(value: T): Bindable<T>;
  * @syscap SystemCapability.ArkUI.ArkUI.Full
  * @since 20
  */
-export declare function applyStyles<T extends CommonMethod>(this: T, customStyles: CustomStyles): T;
+export declare function applyStyles<T extends CommonMethod>(self: T, customStyles: CustomStyles): T;
diff --git a/api/arkui/component/customComponent.d.ets b/api/arkui/component/customComponent.d.ets
index cfd065cf4..64f7e4cc2 100644
--- a/api/arkui/component/customComponent.d.ets
+++ b/api/arkui/component/customComponent.d.ets
@@ -168,7 +168,7 @@ export declare abstract class BaseCustomDialog<T extends BaseCustomDialog<T, T_O
      * @since 20
      */
     @ComponentBuilder
-    static $_instantiate(
+    static $_instantiate<T, T_Options>(
         factory: () => T,
         initializers?: T_Options,
         @Builder content?: () => void
@@ -229,7 +229,7 @@ export declare abstract class CustomComponent<T extends CustomComponent<T, T_Opt
      * @since 20
      */
     @ComponentBuilder
-    static $_instantiate(
+    static $_instantiate<T, T_Options>(
         factory: () => T,
         initializers?: T_Options,
         reuseId?: string,
@@ -271,7 +271,7 @@ export declare abstract class CustomComponentV2<T extends CustomComponentV2<T, T
      * @since 20
      */
     @ComponentBuilder
-    static $_instantiate(
+    static $_instantiate<T, T_Options>(
         factory: () => T,
         initializers?: T_Options,
         reuseId?: string,
diff --git a/api/global/resource.d.ts b/api/global/resource.d.ts
index 21a5ebe31..23a71edb9 100644
--- a/api/global/resource.d.ts
+++ b/api/global/resource.d.ts
@@ -96,14 +96,14 @@ export interface Resource {
   /**
    * Set params.
    *
-   * @type { ?Array<Object | undefined> }
+   * @type { ?Array<String> }
    * @syscap SystemCapability.Global.ResourceManager
    * @crossplatform
    * @atomicservice
    * @since 20
    * @arkts 1.2
    */
-  params?: Array<Object | undefined>;
+  params?: Array<String>;
 
   /**
    * Set params.
