diff --git a/api/arkui/component/distortionComponent.d.ets b/api/arkui/component/distortionComponent.d.ets
index a5b07a3..33e19ca 100644
--- a/api/arkui/component/distortionComponent.d.ets
+++ b/api/arkui/component/distortionComponent.d.ets
@@ -112,7 +112,8 @@ export declare interface DistortionComponentOptions {
  * @stagemodelonly
  * @since 26.0.0
  */
-export declare interface DistortionComponentAttribute extends CommonMethod {}
+// REASON: DistortionComponent is not implemented in arkui
+// export declare interface DistortionComponentAttribute extends CommonMethod {}
 
 /**
  * Defines a DistortionComponent that provides spatial distortion visual effects.
@@ -125,9 +126,10 @@ export declare interface DistortionComponentAttribute extends CommonMethod {}
  * @stagemodelonly
  * @since 26.0.0
  */
-@memo
-@ComponentBuilder
-export declare function DistortionComponent(
-    content: ComponentContent,
-    options?: DistortionComponentOptions,
-): DistortionComponentAttribute;
+// REASON: DistortionComponent is not implemented in arkui
+// @memo
+// @ComponentBuilder
+// export declare function DistortionComponent(
+//     content: ComponentContent,
+//     options?: DistortionComponentOptions,
+// ): DistortionComponentAttribute;
