diff --git a/api/arkui/component/richEditor.d.ets b/api/arkui/component/richEditor.d.ets
index a265202..d894ea6 100644
--- a/api/arkui/component/richEditor.d.ets
+++ b/api/arkui/component/richEditor.d.ets
@@ -2357,7 +2357,8 @@ export declare interface RichEditorAttribute extends CommonMethod {
      * @stagemodelonly
      * @since 26.0.0
      */
-    default horizontalScrolling(enabled: boolean | undefined): this;
+    // Reason: not implemented in arkui
+    // default horizontalScrolling(enabled: boolean | undefined): this;
 }
 /**
  * the callback of cut event.
