diff --git a/api/arkui/component/text.d.ets b/api/arkui/component/text.d.ets
index 724d3cd..62fefbc 100644
--- a/api/arkui/component/text.d.ets
+++ b/api/arkui/component/text.d.ets
@@ -1023,7 +1023,8 @@ export declare enum MarqueeState {
      * @syscap SystemCapability.ArkUI.ArkUI.Full
      * @since 23
      */
-    FINISH = 2
+    FINISH = 2,
+    STOP = 3,
 }
 /**
  * Defines marquee start policy.
